File: Final/Code/Project_3/GeneratingFunctions.cpp
#include <iostream>
using namespace std;

int count_coin_combinations(int total) {
    int count = 0;
    for (int c = 0; c <= total / 5; ++c) {
        int remain_after_5 = total - 5 * c;
        for (int b = 0; b <= remain_after_5 / 2; ++b) {
            int a = remain_after_5 - 2 * b;
            if (a >= 0) {
                count++;
                cout << "(a=" << a << ", b=" << b << ", c=" << c << ")" << endl;
            }
        }
    }
    return count;
}

int main() {
    int total;
    cout << "Nhập n: ";
    cin >> total;
    int result = count_coin_combinations(total);
    cout << "Tổng số cách chọn đồng xu để được " << total << ": " << result << endl;
    return 0;
}

File: Final/Code/Project_3/GeneratingFunctions.py
def count_coin_combinations(total):
    count = 0
    # c: số đồng xu mệnh giá 5
    for c in range(total // 5 + 1):
        remain_after_5 = total - 5 * c
        # b: số đồng xu mệnh giá 2
        for b in range(remain_after_5 // 2 + 1):
            a = remain_after_5 - 2 * b
            # a phải là số nguyên không âm
            if a >= 0:
                count += 1
                print(f"(a={a}, b={b}, c={c})")
    return count

total = int(input("Nhập n: "))
result = count_coin_combinations(total)
print(f"Tổng số cách chọn đồng xu để được {total}: {result}")
